### PR TITLE
BF: Fixed the logic in checking that ref_ds param is in the correct range

### PR DESCRIPTION
**Pull request by swaroopgj**
Also added a unittest.

### CODE REVIEW

File: mvpa2/algorithms/hyperalignment.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def train(self, datasets):
             ref_ds = np.argmax(nfeatures)
         else:
             ref_ds = params.ref_ds
-            if ref_ds < 0 and ref_ds >= ndatasets:
+            if ref_ds >= ndatasets:
```
Comment (nno) [inline mvpa2/algorithms/hyperalignment.py]: I like that it gives an informative error message for a too large value of ref_ds, but what about giving a non-integer or negative value?

Comment (swaroopgj) [inline mvpa2/algorithms/hyperalignment.py]: Good point. Those two check are done during declaration in Parameter() as EnsureInt() and EnsureRange(min=0)

Comment (nno) [inline mvpa2/algorithms/hyperalignment.py]: Ok, very good, thanks. How about adding a comment indicating that the Parameter class takes care of this? It may also be good to add two other cases to the test_hyper_ref_ds_range_checks function, using negative and non-integer ref_ds values.

Comment (swaroopgj) [inline mvpa2/algorithms/hyperalignment.py]: Current behavior accepts fractions that can be converted into non-negative ints (EnsureInt()). Would it be better to force the parameter to be of dtype int instead?

**Comment by swaroopgj** [comment]
Any more comments on this fix?

**Comment by nno** [comment]
- the current test code tests for an exception when ``__call__`` is used on an untrained Hyperalignment instance with ref_ds out of range. I haven't tried this but it seems an exception may be raised because the Hyperalignment instance is not trained, not because ref_ds is out of range. In any case, it would be good to have a test case where it checks an exception is thrown after training but with an out-of-range ref_ds value.
- could you add test cases verifying an exception is thrown negative and non-integer ref_ds

**Comment by swaroopgj** [comment]
This exception should happen before training, which it does.
Negative and non-integer values are handled by Parameter class and it tries to cast them as Int with `EnsureInt()` if possible. I added a few lines to assert that.

**Comment by yarikoptic** [comment]
better avoid magical casts (e.g. -0.5 -> 0), just give it ```EnsureInt(min=0)```

**Comment by swaroopgj** [comment]
There is no such argument, but I go that functionality by switching constraints order. Also changed the test accordingly.
